iothub_client/src/iothub_client_ll.c

```diff
@@ -1575,6 +1587,23 @@ IOTHUB_CLIENT_RESULT IoTHubClient_LL_SetOption(IOTHUB_CLIENT_LL_HANDLE iotHubCli
                 result = IOTHUB_CLIENT_OK;
             }
         }
+        else if (strcmp(optionName, OPTION_DIAGNOSTIC_SAMPLING_PERCENTAGE) == 0)
+        {
+            uint32_t percentage = *(uint32_t*)value;
+            if (percentage < 0 || percentage > 100)
```

Comment:
You should not check for negative, this is an unsigned int.

Reply:
done